owner-onboarding-server/src/ov_management/mod.rs

```diff
@@ -0,0 +1,256 @@
+    }
+
+    let mut unowned_guid = Vec::new();
+    let mut owned_guid = Vec::new();
+    for (idx,ov) in ovs.iter().enumerate() {
+        match validate_ov(ov,&user_data.owner_key) {
+            Ok(state) => {
+                if state {           
+                    owned_guid.push(ov.header().guid().to_string());
+                } else {
+                    unowned_guid.push(ov.header().guid().to_string());
+                }
+            }
+            Err(e) => match e {
+                ErrorCode::IncompleteVoucherDescription(_) =>
+                    return ErrorMsg::generate_with_description(ErrorCode::IncompleteVoucher,
+                    owned_guid.len(),
+                    &e.to_string()),
+                ErrorCode::InvalidVoucherSignatureDescription(_) =>
+                    return ErrorMsg::generate_with_description(ErrorCode::InvalidVoucherSignature,
+                    idx,
+                    &e.to_string()),
+                ErrorCode::PemParseErrorDescription(_) =>
+                    return ErrorMsg::generate_with_description(ErrorCode::PemParseError,
+                    owned_guid.len(),
+                    &e.to_string()),
+                _ => {}
+            }
+        }
+    }
+    
+    if !unowned_guid.is_empty() {
+        return ErrorMsg::generate(ErrorCode::UnwonedVoucher, unowned_guid);
+    }
+    
+    for ov in ovs.iter() {
+            let _ = user_data
+                .ownership_voucher_store
+                .store_data(ov.clone().header().guid().to_owned(), ov.clone())
+                .await;
+        }
+
+    let uuid = warp::reply::json(&owned_guid);
+    Ok(Box::new(warp::reply::with_status(uuid, StatusCode::CREATED)))
+
+}
+
+async fn delete_ov(
+    uuids: Vec<String>,
+    user_data: Arc<OwnerServiceUD>,
+) -> core::result::Result<Box<dyn warp::reply::Reply>,warp::Rejection> 
+{
+    let mut unknown_device= Vec::new();
+    for uuid in &uuids {
+        if Guid::from_str(uuid).is_err() {
+            return ErrorMsg::generate(ErrorCode::ParseError, "Invalid data");
+        }
+        match user_data
+        .ownership_voucher_store
+        .load_data(&Guid::from_str(uuid).unwrap())
+        .await
+        {
+            Ok(Some(_)) => {},
+            Ok(None) => {
+                unknown_device.push(uuid.to_string());
+            }
+            Err(_) => {},
+        }; 
+    }
+
+    if !unknown_device.is_empty() {
+        return  ErrorMsg::generate(ErrorCode::UnknownDevice, unknown_device);
+    }
+
+    for i in uuids {
+        user_data
+        .ownership_voucher_store
+        .destroy_data(&Guid::from_str(&i).unwrap())
+        .await.unwrap();
+    }
+
+    Ok(Box::new(warp::reply()))
+}
+
+
+const ACCEPTED_CONTENT_TYPE:[&str; 2] = ["application/x-pem-file","application/cbor"];
+
+#[derive(Clone)]
+#[derive(Serialize,Deserialize)]
+#[derive(Eq,PartialEq,Hash)]
+#[derive(Error,Debug)]
+enum ErrorCode {
+    #[error("error")]
```

Comment:
The argument to the `#[error()]` call should be an actual error description.
For example: `#[error("Parse error: {0}")]`.
It would also be great to actually pass the error message along with this perhaps, like `ParseError(crate::Error)`, so that that information is passed along via this.

Doing both of those, you could get mostly rid of the `generate` and `generate_with_Description` methods that take extra arguments, and instead just have the `ErrorCode` struct generate those from the value.